### DISCUSSION TITLE
`signInwithOauth()` redirectTo URL is ignored by supabase.

### DISCUSSION THREAD

**Discussion by jjfantini** [supabase/supabase#27991] [Questions]
I am using the suggested method of setting the redirect URL in the authentication method as shown here: https://supabase.com/docs/guides/auth/redirect-urls#vercel-preview-urls

```typescript
import { createClient } from '@/utils/supabase/client'
import { useRouter } from 'next/navigation'
import { socialLoginRedirectPath } from '../constants'



export function useLogin() {
  const router = useRouter()

  const handleLoginWithOauth = (
    provider:
      | 'google'
      | 'github'
      | 'twitter'
      | 'discord'
      | 'linkedin_oidc'
      | 'apple'
  ) => {
    return async function() {

      const getURL = () => {
        let url =
          process?.env?.NEXT_PUBLIC_SITE_URL ?? // Set this to your site URL in production env.
          process?.env?.NEXT_PUBLIC_VERCEL_URL ?? // Automatically set by Vercel.
          'http://localhost:3000/'
        // Make sure to include `https://` when not localhost.
        url = url.startsWith('http') ? url : `https://${url}`
        // Make sure to include a trailing `/`.
        url = url.endsWith('/') ? url : `${url}/`
        return url
      }

      const supabase = createClient()
      const { data } = await supabase.auth.getUser()

      const { searchParams } = new URL(window.location.href)
      const from = searchParams.get('from')
      const options = {
        redirectTo:
          getURL() +
          socialLoginRedirectPath +
          (from ? `?from=${from}` : ''),
      }

      if (data?.user?.is_anonymous) {
        await supabase.auth.linkIdentity({
          provider,
          options,
        })
        router.replace('/dashboard/home')
      } else {
        const { error } = await supabase.auth.signInWithOAuth({
          provider: provider,
          options,
        })
        if (error) {
          router.replace('/login?message=' + error.message)
        }
      }
    }
  }

  return {
    handleLoginWithOauth,
  }
}
```

It seems that when I am logging in with Google, the app redirects me to the Default Site URL in project settings. So if I run my NextJS app on localhost, it redirects me to the vercel deployment (if that URL iscallback request, it seems to be headers to the  set). Even if I send different ore them. If i make the Site URl localhost, the vercel deployment redirects to a localhost server that isnt running.

The project I am working on is open-source, so you can check out the code if you would like. Just ping me in the replies.

**Comment by GaryAustin1** [comment]
You don't mention if you set your Additional Redirect URLs in the dashboard.  They have to be full URL matches (wildcards available) to the redirects you provide in your call.

**Reply by tokarskyibogdan** [reply]
I'm facing the same issue, and even if I set direct URLs in admin panel they are ignored

**Reply by pierroo** [reply]
did you guys find a solution for that?
It will only care about my siteUrl, and completely ignore additional redirect urls.
(which means I cannot dev / test on localhost unless I put my siteUrl as localhost, which I dont want. And even then, it forces the port 3000 which is not correct for me)

so redirectTo is completely ignored for signInWithOAuth.

**Reply by ngoldbla** [reply]
I have the same problem and would like to find a solution

**Reply by JuliGame** [reply]
Same issue here. Auth url configurations has the proper url. Its being ignored and redirecting to Site URL instead.

**Reply by slang21** [reply]
Also having this issue.

**Reply by GaryAustin1** [reply]
This almost always is the redirectTo not matching the redirect URL's in the dashboard EXACTLY  or a setup issue with OAuth and the various URLs in settings for the OAuth provider.

Just saying you have the same issue does not make anything happen.  There are no known bugs that I know of and people using are not all of a sudden saying it is broken.

You might try Discord for a more interactive help and post specifics on your call and URL settings.

**Reply by slang21** [reply]
In my case the issue was that it was not an EXACT match, but this seems to be a bug with the wildcard. Opened an issue [here](https://github.com/supabase/supabase/issues/34912)
